pagination/src/main/java/com/cerner/beadledom/pagination/OffsetPaginationModule.java

```diff
@@ -0,0 +1,83 @@
+package com.cerner.beadledom.pagination;
+
+import com.cerner.beadledom.pagination.models.OffsetPaginationConfiguration;
+import com.cerner.beadledom.pagination.parameters.LimitParameter;
+import com.cerner.beadledom.pagination.parameters.OffsetParameter;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.name.Named;
+import javax.inject.Singleton;
+
+/**
+ * Module to provide the necessary configurations for using Offset Based Pagination.
+ *
+ * <p>Module provides the configurations needed to use offset based pagination. This includes
+ * the {@link OffsetPaginatedListLinksWriterInterceptor} as well as default values for the limit and
+ * offset fields and their respective field names. The default values for the limit, offset, and
+ * their respective field names can be configured using one of the provided constructors.
+ *
+ * @author Will Pruyn
+ * @author Ian Kottman
+ * @since 3.1
+ */
+public class OffsetPaginationModule extends AbstractModule {
+  private final OffsetPaginationConfiguration config;
+
+  /**
+   * Default constructor that sets the configurable fields using the Beadledom defaults. For
+   * the default values see {@link OffsetPaginationConfiguration}
+   */
+  public OffsetPaginationModule() {
+    this.config = OffsetPaginationConfiguration.builder().build();
+  }
+
+  /**
+   * Override Beadledom defaults.
+   */
+  public OffsetPaginationModule(OffsetPaginationConfiguration config) {
+    this.config = config;
+  }
```

Comment:
I feel like a cleaner approach would be to have a requireBinding for this OffsetPaginationConfiguration which the consumer would have to provide. This way, the module doesn't have to worry about handing defaults vs overrides.

I think you just need to add the following in the configure() method below
```
@Override
protected void configure() {
  requireBinding(OffsetPaginationConfiguration.class);
  ........
  ........
}
```

This way we can get rid off the constructors altogether.  If you choose to do this, it would also require a README update.

Reply:
Might be able to use an optional binding for this with a default.

Reply:
Even better!